spec/lib/jobs/create-default-pollers-spec.js

```diff
@@ -143,13 +159,44 @@ describe("Job.Pollers.CreateDefault", function () {
                 .to.have.property('node', nodeId);
             expect(waterline.catalogs.findMostRecent.secondCall.args[0])
                 .to.have.property('source', 'snmp-1');
-            expect(waterline.workitems.findOrCreate).to.have.been.calledTwice;
+            expect(waterline.workitems.findOrCreate).to.have.been.calledThrice;
             expect(waterline.workitems.findOrCreate).to.have.been.calledWith(
                 { node: nodeId, config: { command: pollers[0].config.command } }, pollers[0]);
             expect(waterline.workitems.findOrCreate).to.have.been.calledWith(
                 { node: nodeId, config: { command: pollers[1].config.command } }, pollers[1]);
         });
     });
+    
+    
+    it('should fail on missing redfish service', function () {
+        var nodeId = 'bc7dab7e8fb7d6abf8e7d6ad';
+        pollers = [{
+            "type": "redfish",
+            "pollInterval": 60000,
+            "config": {
+                "command": "thermal"
+            }
+        }];
+        
+        waterline.nodes = {
+            needByIdentifier: sinon.stub().resolves({
+                obmSettings: []
+            })
+        };
+
+        var job = new CreateDefaultPollers(
+            { nodeId: nodeId, pollers: pollers },
+            { graphId: uuid.v4() },
+            uuid.v4()
+        );
+
+        return job.run()
+        .catch(function(err) {
```

Comment:
How about the `job.run()` is fulfilled not rejected? so that your assertion logic will not be checked but the unit-test will pass. So you should explicitly fail the test case if the job.run is fulfilled. Below is an example:
https://github.com/RackHD/on-tasks/blob/master/spec%2Flib%2Futils%2Fjob-utils%2Fracadm-tool-spec.js#L156-L164

If no complex assertion logic in catch block, we should using following line to check the promise is rejected, this is the safe way and it will fail if `job.run` is fulfilled:
```
return job.run().to.be.rejectedWith(Error, 'xxxxx');
```